### ISSUE TITLE
SFileOpenArchive fails, GetLastError() = 13, for two starcraft maps that play correctly in-game

### ISSUE THREAD

**Issue by zzlk** [ladislav-zezula/StormLib#373]
Hi,

using this version:

```
commit e01d93cc8ae743cfe2da5450854c5d2e3a939265 (upstream/master)
Author: Ladislav Zezula <...>
Date:   Fri Dec 6 16:23:35 2024 +0100

    Replaced afxres.h with winres.h
```

I have two maps that fail to open:

[err13_maps.zip](https://github.com/user-attachments/files/18156703/err13_maps.zip)

```
SFileOpenArchive. GetLastError: 13, filename: Mission22.scx
SFileOpenArchive. GetLastError: 13, filename: Apogée Zerg (6).scx
```

they do parse/run ingame (I also tested them in game, they run in SC remastered, I haven't checked 1.16).

![image](https://github.com/user-attachments/assets/0dae7fa3-f1a8-46a7-beac-05884d0fd196)
![image](https://github.com/user-attachments/assets/189327ff-004c-4562-b50d-88b864a7a472)

**Comment by ladislav-zezula** [comment]
They do open pretty well, both in MPQEditor and StormLib test program.
![image](https://github.com/user-attachments/assets/7dacc3f3-0e70-4c82-b9d7-b0de49ee64cb)

**Comment by zzlk** [comment]
I did some debugging and I figured out the issue.

The issue is that the maps are read only
```
ls -ltrp
total 108
-r--r--r-- 1 30646 Aug 21  1999  Mission22.scx
-r--r--r-- 1 73885 Nov  1  2001 'Apogée Zerg (6).scx'
```

I was also not passing STREAM_FLAG_READ_ONLY to SFileOpenArchive. Once I do that, it works fine.

The error comes from `BaseFile_Open` in the linux section:

```c++
        handle = open(szFileName, oflag | O_LARGEFILE);
        if(handle == -1)
        {
            fprintf(stderr, "2.5. BaseFile_Open: %s, errno = %d, oflag = %d\n", szFileName, errno, oflag);

            pStream->Base.File.hFile = INVALID_HANDLE_VALUE;
            dwLastError = errno;
            return false;
        }
```
on linux, errno = 13 means "permission denied", whereas on windows errcode = 13 means "invalid data" . I guess my confusion came from GetLastError() returning errno codes instead of winapi codes. I guess it makes sense since I am using the GetLastError that is exported by stormlib, but yeah, a bit confusing.

I think this issue is solved and I guess there's nothing that stormlib needs to do here. Maybe better naming/documentation for GetLastError()? But I think that would be about it.

Thanks for checking.
